### ISSUE TITLE
Could implement DoubleEndedIterator

### ISSUE THREAD

**Issue by alecmocatta** [stephaneyfx/enum-iterator#9]
I need the variants in reverse order and, rather than collecting and then iterating, it would be handy if [`std::iter::DoubleEndedIterator`](https://doc.rust-lang.org/stable/std/iter/trait.DoubleEndedIterator.html) was implemented such that I could do `Enum::into_enum_iter().rev()`.

Relatedly, [`FusedIterator`](https://doc.rust-lang.org/stable/std/iter/trait.FusedIterator.html) can also be implemented.

**Comment by stephaneyfx** [comment]
Do you need `DoubleEndedIterator` for enumerations with data (i.e. with variants with fields) and structures? Or only for C-like enumerations?

**Comment by alecmocatta** [comment]
My specific use case right now is with C-like enumerations. And this is my workaround:

```rust
for x in Enum::into_enum_iter().collect::<Vec<_>>().into_iter().rev() { ... }
```

**Comment by stephaneyfx** [comment]
Thank you. I am working on it.

**Comment by stephaneyfx** [comment]
`1.0.0` was published and provides an iterator to iterate over variants in reverse order. There is no implementation of `DoubleEndedIterator` though. Rather, two separate iterator types are provided to iterate forward or backward. Based on your previous messages, the reverse-order iterator should solve your issue.

Note that the API provided by this crate changed significantly. Notably the main trait is now called `Sequence`. The latest stable Rust (1.61.0) is needed. Below is an example:

```rust
use enum_iterator::{reverse_all, Sequence};

#[derive(Debug, PartialEq, Sequence)]
enum Color { Red, Green, Blue }

assert_eq!(
    reverse_all::<Color>().collect::<Vec<_>>(),
    [Color::Blue, Color::Green, Color::Red],
);
```

Does this solve your issue?

**Comment by stephaneyfx** [comment]
The original issue of iterating backwards without collecting was resolved in `1.0.0`.
